x-pack/plugin/migrate/src/main/java/org/elasticsearch/system_indices/task/SystemIndexMigrator.java

```diff
@@ -467,38 +457,43 @@ private void migrateSingleIndex(ClusterState clusterState, Consumer<BulkByScroll
         }
     }
 
-    private void createIndex(SystemIndexMigrationInfo migrationInfo, ActionListener<ShardsAcknowledgedResponse> listener) {
+    private void createIndex(SystemIndexMigrationInfo migrationInfo, ActionListener<CreateIndexResponse> listener) {
         logger.info("creating new system index [{}] from feature [{}]", migrationInfo.getNextIndexName(), migrationInfo.getFeatureName());
 
-        final CreateIndexClusterStateUpdateRequest createRequest = new CreateIndexClusterStateUpdateRequest(
-            "migrate-system-index",
-            migrationInfo.getNextIndexName(),
-            migrationInfo.getNextIndexName()
-        );
-
+        CreateIndexRequest createIndexRequest = new CreateIndexRequest(migrationInfo.getNextIndexName());
         Settings.Builder settingsBuilder = Settings.builder();
         if (Objects.nonNull(migrationInfo.getSettings())) {
             settingsBuilder.put(migrationInfo.getSettings());
             settingsBuilder.remove("index.blocks.write");
             settingsBuilder.remove("index.blocks.read");
             settingsBuilder.remove("index.blocks.metadata");
         }
-        createRequest.waitForActiveShards(ActiveShardCount.ALL)
-            .mappings(migrationInfo.getMappings())
+        createIndexRequest.cause(SystemIndices.MIGRATE_SYSTEM_INDEX_CAUSE)
+            .ackTimeout(TimeValue.ZERO)
+            .masterNodeTimeout(TimeValue.MINUS_ONE)
+            .waitForActiveShards(ActiveShardCount.ALL)
+            .mapping(migrationInfo.getMappings())
             .settings(Objects.requireNonNullElse(settingsBuilder.build(), Settings.EMPTY));
-        metadataCreateIndexService.createIndex(TimeValue.MINUS_ONE, TimeValue.ZERO, null, createRequest, listener);
+
+        baseClient.admin().indices().create(createIndexRequest, listener);
     }
 
-    private void createIndexRetryOnFailure(SystemIndexMigrationInfo migrationInfo, ActionListener<ShardsAcknowledgedResponse> listener) {
+    private void createIndexRetryOnFailure(SystemIndexMigrationInfo migrationInfo, ActionListener<CreateIndexResponse> listener) {
         createIndex(migrationInfo, listener.delegateResponse((l, e) -> {
-            logger.warn("createIndex failed, retrying after removing index [{}] from previous attempt", migrationInfo.getNextIndexName());
+            logger.warn(
```

Comment:
Nice improvement. I wonder if we should log the whole error stack trace  with a `warn.(message, error)` method?